pallets/restricted-xtokens/src/tests.rs

```diff
@@ -0,0 +1,538 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+#![cfg(test)]
+
+use cumulus_primitives_core::ParaId;
+use frame_support::{assert_noop, assert_ok, traits::Currency};
+use mock::*;
+use orml_traits::MultiCurrency;
+use sp_runtime::{traits::AccountIdConversion, AccountId32};
+use xcm_simulator::TestExt;
+
+use super::*;
+use crate::tests::para::RESTRICTON_TRIGGERED;
+fn para_a_account() -> AccountId32 {
+	ParaId::from(1).into_account_truncating()
+}
+
+// Not used in any unit tests, but it's super helpful for debugging. Let's
+// keep it here.
+#[allow(dead_code)]
+fn print_events<Runtime: frame_system::Config>(name: &'static str) {
+	println!("------ {:?} events -------", name);
+	frame_system::Pallet::<Runtime>::events()
+		.iter()
+		.for_each(|r| println!("> {:?}", r.event));
+}
+
+mod para_a_to_relay {
+	use super::*;
+
+	#[test]
+	fn restrict_normal_transfer_relay_currency() {
+		TestNet::reset();
```

Comment:
These are tests that are using the `xcm-simulator` from what I can tell. Given our efforts to clean up the integration test setup, I'm not sure how I feel about having these tests here in the first place, and re-using `xcm-simulator`.,,

Reply:
Agreed, but I also not so keen on bringing fudge in here for unit-testing. I can remove all tests and just keep integration testm wdyt?

Reply:
Yup, I think that's a good move.

Reply:
I think we can just add some of the missing error cases to the integration tests and just remove the ones here.

Reply:
Not sure how possible this is given the overhead. I would be fine with having just the restrict and allow calls there. Or what cases do you see missing?